core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala

```diff
@@ -199,14 +201,20 @@ private[spark] class HeartbeatReceiver(sc: SparkContext, clock: Clock)
       if (now - lastSeenMs > executorTimeoutMs) {
         logWarning(s"Removing executor $executorId with no recent heartbeats: " +
           s"${now - lastSeenMs} ms exceeds timeout $executorTimeoutMs ms")
-        scheduler.executorLost(executorId, SlaveLost("Executor heartbeat " +
-          s"timed out after ${now - lastSeenMs} ms"))
```

Comment:
Since remove executor in `CoarseGrainedSchedulerBackend` will always do `scheduler.executorLost()`, so it's no longer needed here.

Reply:
what about other `SchedulerBackend`? We only send `RemoveExecutor` event for `CoarseGrainedSchedulerBackend`

Reply:
Hmm...This may can be a problem...

Reply:
what if we don't remove it?

Reply:
If we fail to remove it from `CoarseGrainedSchedulerBackend`, then, it means the executor has been removed previously(e.g. `onDisconnected()`). And `scheduler.executorLost()` must has been called, too.

Reply:
shall we still do this for non-`CoarseGrainedSchedulerBackend`?

Reply:
I think this is needed for non-`TaskSchedulerImpl` ?

This can be a behavior change for other TaskScheduler, I'll revert this later.